src/me/mrCookieSlime/Slimefun/listeners/ArmorListener.java

```diff
@@ -32,48 +34,73 @@ public ArmorListener(SlimefunPlugin plugin) {
 		plugin.getServer().getPluginManager().registerEvents(this, plugin);
 	}
 	
-	@EventHandler(priority=EventPriority.MONITOR)
+	@EventHandler(priority = EventPriority.MONITOR)
 	public void onDamage(EntityDamageEvent e) {
 		if (e.getEntity() instanceof Player && !e.isCancelled()) {
 			Player p = (Player) e.getEntity();
 			for (ItemStack armor: p.getInventory().getArmorContents()) {
-				if (armor != null && SlimefunItem.getByItem(armor) != null) {
-					if (SlimefunItem.getByItem(armor).isItem(SlimefunItems.ENDER_BOOTS) && Slimefun.hasUnlocked(p, SlimefunItems.ENDER_BOOTS, true)) {
-						if (e instanceof EntityDamageByEntityEvent && ((EntityDamageByEntityEvent) e).getDamager() instanceof EnderPearl) e.setCancelled(true);
+				SlimefunItem item = SlimefunItem.getByItem(armor);
+				if (armor != null && item != null) {
+					if (item.isItem(SlimefunItems.ENDER_BOOTS)
```

Comment:
Comparing the ItemStacks seems a bit harsh, you should just compare the id instead, much lighter.

Reply:
Is there a good way of getting the ID for things like Ender Boots here? Looking around it seems the IDs are hardcoded in SlimefunSetup and aren't easily derived from anywhere.

Reply:
Sadly not, maybe we could have a utility class for that...
Too much work though.

For now it's fine to hard-code it in. After all they haven't changed since years and probably never will but I know it's a bad design.